superset/databases/commands/test_connection.py

```diff
@@ -145,7 +152,7 @@ def ping(engine: Engine) -> bool:
             )
             # check for custom errors (wrong username, wrong password, etc)
             errors = database.db_engine_spec.extract_errors(ex, context)
-            raise DatabaseTestConnectionFailedError(errors) from ex
+            raise SupersetErrorsException(errors) from ex
```

Comment:
@Antonio-RiveroMartnez I'm looking into some 500 errors for test connections, and I saw this change. What would happen if I were to change this back to raise the `DatabaseTestConnectionFailedError` which is a 422? Would it break anything from this PR? I checked through the implementation and it looks like it should still catch the SupersetErrorsExceptions because it is the base class, but I wanted to see if I might be missing anything here.

Reply:
Hey @eschutho, the problem of doing that is that we would lost the `original message` and instead the message from DatabaseTestConnectionFailedError would be used (`Connection failed, please check your connection settings`) and not the custom message with roles after parsing the original message.